src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -181,6 +184,28 @@ class Http2Connection  {
      *  and SSLTube.recycler.
      */
 
+    // An Idle connection is one that has no active streams
+    // and has not sent the final stream flag
+    final class IdleConnectionTimeoutEvent extends TimeoutEvent {
+
+        IdleConnectionTimeoutEvent(Duration duration) {
+            super(duration);
+        }
+
+        @Override
+        public void handle() {
+            if (debug.on()) {
+                debug.log("HTTP connection idle for too long");
+            }
+            HttpTimeoutException hte = new HttpTimeoutException("HTTP connection idle, no active streams. Shutting down.");
```

Comment:
Hello Conor,

The javadoc of `HttpTimeoutException` states "Thrown when a response is not received within a specified time period.", which isn't what we are using it for, here. Do you think we should instead just use a `Exception` (or some internal exception type) since this (as far as I can see) won't get propagated to the application?

Reply:
At the point were this exception is generated, it shouldn't be propagated anywhere, because there should be no open stream on that connection. However, we could have some hidden race conditions (even if that's not supposed to happen), where the connection would have been found in the pool by a new request, where the connection gets closed as idle at the same time where the request attempts to open a stream on the connection. In that case, the exception may be propagated, in which case `HttpTimeoutException` (hmmm, or should it actually be `HttpConnectTimeoutException`?) will be propagated to the request - which should allow the stack (or at worst the caller?) to retry.

Reply:
I think `HttpConnectTimeoutException` sounds like a more relevant exception, even if its javadoc has reference to `... is not successfully established within a specified time period.`